src/pytorch_lightning/core/module.py

```diff
@@ -175,12 +179,13 @@ def lr_schedulers(self) -> Optional[Union[LRSchedulerTypeUnion, List[LRScheduler
 
     @property
     def trainer(self) -> "pl.Trainer":
-        if not self._running_torchscript and self._trainer is None:
+        if self._running_torchscript and self._trainer is not None:
```

Comment:
inverted logic to correct:

```sh
src/pytorch_lightning/core/module.py:188: error: Incompatible return value type (got "Optional[Trainer]", expected "Trainer")  [return-value]
```

Reply:
But it's not correct. Most of the time we are not running with torchscript but we have the trainer reference not None. 

I guess it should be

```py
        if self._running_torchscript or self._trainer is None:
            raise RuntimeError(...)
        return self._trainer
```

Reply:
If left as the original, the mypy error will raise. Should I just ignore the error?

Reply:
If you apply my suggestion, there won't be an error. I couldn't see one.